src/platform_impl/macos/app_state.rs

```diff
@@ -265,8 +281,9 @@ impl AppState {
         HANDLER.set_in_callback(false);
     }
 
-    pub fn wakeup() {
-        if !HANDLER.is_ready() {
+    pub fn wakeup(panic_info: Weak<PanicInfo>) {
+        let panic_info = panic_info.upgrade().unwrap();
```

Comment:
If these `unwrap`s are more-or-less unreachable, then making them `expect`s that indicate that this panicking is a developer error / invariant violation would be great.